src/ui/components/ErrorBoundary.tsx

```diff
@@ -28,9 +28,9 @@ export default class ErrorBoundary extends React.Component<ErrorBoundaryProps, E
 
     static getDerivedStateFromError = (error: Error) => ({ hasErr: true, errText: error.message });
 
-    render() {
-        if (!this.state.hasErr) return this.props.children;
```

Comment:
you can just use hooks in here no? I did it on Vencord a few days ago

Reply:
iirc Vencord exports a custom class component that lets you call hooks in the render() method

Reply:
nevermind im not sure

Reply:
stop waffling you don't know what you're talking about

Reply:
you can probably call a hook there anyways I did it without any extras in a Discord class-based component

Reply:
bestie you cant call hooks in regular react class components

Reply:
bestie I literally did it yesterday xo

Reply:
Rosie is right, you can only use it in function components